src/Sylius/Component/Product/tests/Resolver/DefaultProductVariantResolverTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\Component\Product\Resolver;
+
+use Doctrine\Common\Collections\Collection;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Sylius\Component\Product\Model\ProductInterface;
+use Sylius\Component\Product\Model\ProductVariant;
+use Sylius\Component\Product\Model\ProductVariantInterface;
+use Sylius\Component\Product\Repository\ProductVariantRepositoryInterface;
+use Sylius\Component\Product\Resolver\DefaultProductVariantResolver;
+use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
+
+final class DefaultProductVariantResolverTest extends TestCase
+{
+    /**
+     * @var ProductVariantRepositoryInterface<ProductVariant>&MockObject
+     */
+    private MockObject $productVariantRepository;
+
+    private DefaultProductVariantResolver $defaultProductVariantResolver;
+
+    protected function setUp(): void
+    {
+        $this->productVariantRepository = $this->createMock(ProductVariantRepositoryInterface::class);
+        $this->defaultProductVariantResolver = new DefaultProductVariantResolver($this->productVariantRepository);
+    }
+
+    public function testImplementsVariantResolverInterface(): void
+    {
+        $this->assertInstanceOf(ProductVariantResolverInterface::class, $this->defaultProductVariantResolver);
+    }
+
+    public function testReturnsFirstVariantIfProductHasNoId(): void
+    {
+        $product = $this->createMock(ProductInterface::class);
+        $productVariant = $this->createMock(ProductVariantInterface::class);
+        $variants = $this->createMock(Collection::class);
```

Comment:
The ArrayCollection with variants could be created here instead of mocking Collection